cura/UI/WelcomePagesModel.py

```diff
@@ -0,0 +1,68 @@
+# Copyright (c) 2019 Ultimaker B.V.
+# Cura is released under the terms of the LGPLv3 or higher.
+import os
+from typing import TYPE_CHECKING, Optional, List, Dict, Any
+
+from PyQt5.QtCore import QUrl, Qt
+
+from UM.Qt.ListModel import ListModel
+from UM.Resources import Resources
+
+if TYPE_CHECKING:
+    from PyQt5.QtCore import QObject
+
+
+class WelcomePagesModel(ListModel):
+
+    IdRole = Qt.UserRole + 1  # Page ID
+    PageUrlRole = Qt.UserRole + 2  # URL to the page's QML file
+    NextPageIdRole = Qt.UserRole + 3  # The next page ID it should go to
+
+
+    def __init__(self, parent: Optional["QObject"] = None) -> None:
+        super().__init__(parent)
+
+        self.addRoleName(self.IdRole, "id")
+        self.addRoleName(self.PageUrlRole, "page_url")
+        self.addRoleName(self.NextPageIdRole, "next_page_id")
+
+        self._pages = []  # type: List[Dict[str, Any]]
+
+    # Convenience function to get QUrl path to pages that's located in "resources/qml/WelcomePages".
+    def _getBuiltinWelcomePagePath(self, page_filename: str) -> "QUrl":
+        from cura.CuraApplication import CuraApplication
+        return QUrl.fromLocalFile(Resources.getPath(CuraApplication.ResourceTypes.QmlFiles,
+                                                    os.path.join("WelcomePages", page_filename)))
+
+    def initialize(self) -> None:
```

Comment:
Why is the initialize separate?

Reply:
Because `WelcomePagesModel` is created in `CuraApplication`'s constructor. Separating this makes sure that it's initialized at the right moment instead of directly in the constructor because this class may be extended to include more stuff.

Reply:
Is the constructor not the right moment, then?